packages/webapp/src/elections/api/eden-contract.ts

```diff
@@ -138,55 +146,79 @@ const getVoteDataRows = async (
 
     // TODO: see what real data looks like and real use-cases and see if we need the electionState flag;
     // If not, switch this back to getTableRows()
-    const rawRows = await getTableRawRows<VoteData>(CONTRACT_VOTE_TABLE, opts);
-    // const electionState = rawRows[0][0];
-
-    const rows = rawRows.map((row) => row[1]);
+    const rawRows = await getTableRawRows(CONTRACT_VOTE_TABLE, opts);
 
-    if (!rows.length) {
-        return undefined;
-    }
-
-    return rows;
+    if (rawRows?.[0].length) return rawRows.map((row) => row[1]);
+    return rawRows;
 };
 
 export const getVoteData = getVoteDataRows;
 
 const getCommonDelegateAccountForGroupWithThisMember = (
-    round: number,
-    member: EdenMember
+    roundRequested: number,
+    member: EdenMember,
+    voteData?: VoteData
 ) => {
-    const commonDelegate =
-        member.election_rank > round ? member.account : member.representative;
+    let electionRank = member.election_rank;
+    // if the member has an open voteData record (i.e., still participating), get their election rank from that
+    if (voteData?.member === member.account) {
+        electionRank = voteData.round;
+    }
+
+    let commonDelegate = member.representative;
+    if (electionRank > roundRequested) {
+        commonDelegate = member.account;
+    } else if (electionRank < roundRequested) {
+        throw new Error(
+            "Cannot fetch round participants in round member did not participate in."
+        );
+    }
+
     return isValidDelegate(commonDelegate) ? commonDelegate : "";
```

Comment:
If this is working its way into a useQuery, we should probably throw rather than return "", right? Empty delegate account doesn't have meaning to the rest of the app

Reply:
Agreed. Good catch!